openeogeotrellis/geopysparkdatacube.py

```diff
@@ -1242,13 +1254,28 @@ def resample_spatial(
             #return self.apply_to_levels(lambda layer: layer.tile_to_layout(projection, resample_method))
         return self
 
-    def get_cellsize_x(self):
+    def get_cellsize(self):
         max_level = self.get_max_level()
         extent = max_level.layer_metadata.layout_definition.extent
         currentTileLayout: gps.TileLayout = max_level.layer_metadata.tile_layout
+        return (
+            (extent.xmax - extent.xmin) / (currentTileLayout.tileCols * currentTileLayout.layoutCols),
+            (extent.ymax - extent.ymin) / (currentTileLayout.tileRows * currentTileLayout.layoutRows)
+        )
 
-        currentResolutionX = (extent.xmax - extent.xmin) / (currentTileLayout.tileCols * currentTileLayout.layoutCols)
-        return currentResolutionX
+    def estimate_layer_size_in_pixels(self):
```

Comment:
what about taking number of bands into account as well?

Reply:
Should not needed here. I added a comment `The repartitioning only considers the resolution change. It considers that the partitioning took already into account band count and the pixel type, sparse/not, ...`